The KDE implementation doesn't have support to spawn of a terminal and run a gdb retrace session.
This patch implements it.
--- a/kde/apport-kde
+++ b/kde/apport-kde
@@ -16,6 +16,10 @@
 import sys
 import os
 
+# temp until I get a native python way, just too lazy
+from gi.repository import GLib
+import subprocess
+
 try:
     import apport
     from PyQt5.QtCore import (QByteArray,
@@ -361,6 +365,21 @@
             else:
                 QTreeWidgetItem(keyitem, [_('(binary data)')])
 
+    def ui_run_terminal(self, command):
+        terminals = ['x-terminal-emulator', 'konsole', 'yakuake']
+
+        program = None
+        for t in terminals:
+            program = GLib.find_program_in_path(t)
+            if program:
+                break
+
+        if not command:
+            # test mode
+            return 'DISPLAY' in os.environ and program is not None
+
+        subprocess.call([program, '-e', command])
+
     def ui_present_report_details(self, allowed_to_report=True, modal_for=None):
         desktop_info = self.get_desktop_entry()
         self.dialog = ReportDialog(self.report, allowed_to_report, self,
